packages/app/src/visualizer/utils.ts

```diff
@@ -121,21 +118,34 @@ function getSupportedNxVis(
   return NEXUS_VIS[hasMinDims(dataset, 2) ? imageVis : spectrumVis];
 }
 
-function getImplicitDefaultChild(children: Entity[]): Entity | undefined {
-  const groups = children.filter(isGroup);
+function getImplicitDefaultChild(
+  children: Entity[],
+  attrValueStore: AttrValuesStore
+): Entity | undefined {
+  const nxGroups = children
+    .filter(isGroup)
+    .filter((g) => hasAttribute(g, 'NX_class'));
 
   // Look for an `NXdata` child group first
-  const nxDataChild = groups.find((g) => hasNxClass(g, 'NXdata'));
+  const nxDataChild = nxGroups.find(
+    (g) => attrValueStore.getSingle(g, 'NX_class') === 'NXdata'
+  );
```

Comment:
This will trigger requests for attribute values sequentially until it finds the first `NXdata` group. Note that the other `find` loops below won't, since the responses will be cached, so the worst case scenario is when a group has thousands of `NXcollection` children. Since we don't check the `NX_class` of the parent group when looking for an implicit default child, this can potentially happen on any group that contains NX child groups. We could mitigate this by disabling implicit child resolution for non-`NXentry` groups for instance, but I would rather wait for the need to arise in a real use case.

Reply:
Yes, I see how that could become a problem but it seems unlikely. I agree that waiting for the case to actually show up is better than premature optimization.